### DISCUSSION TITLE
Allow to split/group/namespace migrations for FUNCTION

### DISCUSSION THREAD

**Discussion by markus2330** [diesel-rs/diesel#4332] [Ideas]
We have two different kinds of migrations,

1. some that create FUNCTION/TRIGGER (similar to 00000000000000_diesel_initial_setup)
2. some that do the "normal" migrations, creating or alter TABLE etc.

The problem is when there is a bug in the 1. kind. While theoretically it would be possible to simply modify the migrations file, in how the migrations currently works, the bug won't get fixed in existing installations.

The "proper" current way, as I understand it, is to drop the old FUNCTION, and create a new one (with all same but the bug fixed), but in down I would need to restore the old function (with bug). This might make sense if the code is tightly related to the data it is working on (and the bug must be present to work with the old data), but this is massive overkill (duplication of every FUNCTION) if the functions are not related to the data (or work with any version of the migrations).

This use case could be probably solved with #2085, but probably there are more elegant ways to solve the problem with bugs in FUNCTION specifically.

**Comment by weiznich** [comment]
I moved this "bug report" to the discussion forum as it's a question or feature request. The issue tracker is really the wrong place for this as already indicated by the issue template. Also it's discouraged to ask support question on old issues. For future questions: Please neither post them as issue nor as comment below old issues, otherwise this will be just handled as bug report and closed as invalid.

As for the actual question: I might have the capacity to answer that later this week.
